lightning/src/ln/interactivetxs.rs

```diff
@@ -85,15 +87,105 @@ struct NegotiationContext {
 	holder_is_initiator: bool,
 	received_tx_add_input_count: u16,
 	received_tx_add_output_count: u16,
-	inputs: HashMap<SerialId, TxInputWithPrevOutput>,
+	/// The inputs to be contributed by the holder.
+	inputs: HashMap<SerialId, InteractiveTxInput>,
```

Comment:
I've actually thought it would be better to split this into local and remote maps as quite often we're iterating/checking either local / remote. The only time they're combined is during validating and building the final transaction.

I have this change locally for #2981 which I'm pushing up today.

Reply:
Same with the outputs.